revue Post 2100 - ne pas merger

juliettelavoie opened the pull request:
Je fais un PR (pull request) seulement dans le but de pouvoir faire une revue facile du code avec commentaires sur le diff.

Tu peux continuer à faire des commits. 
Il est plus facile pour moi de faire la revue si tu gardes les mêmes noms de fichiers.

!!!! NE PAS MERGE CETTE PR !!!!!

Anne-SophieFortin committed:
Create workflow_ESPO-GE.py

Stage été 2025 - Scénarios Post-2100
But: Post-traiter les simulations CMIP6 allant jusqu'en 2300.

Environnement:
- xscen >= 0.12.3 requis (bug fix des années > 2262)
- Cette version de xscen nécessite une version récente de xclim et celle-ci ne possède plus certaine fonctions qui se retrouvent dans la nouvelle version de xarray (https://github.com/Ouranosinc/xclim/issues/1010). J'ai donc modifié les imports du workflow par rapport à celui de ESPO-G.
- J'utilise xscen-0.12 sur doris

Anne-SophieFortin committed:
Update workflow_ESPO-G.py

Stage été 2025 - Scénarios Post-2100
But: Post-traiter les simulations CMIP6 allant jusqu'en 2300.

Environnement:
- xscen >= 0.12.3 requis (bug fix des années > 2262)
- Cette version de xscen nécessite une version récente de xclim et celle-ci ne possède plus certaine fonctions qui se retrouvent dans la nouvelle version de xarray (Ouranosinc/xclim#1010). J'ai donc modifié les imports du workflow par rapport à celui de ESPO-G.
- J'utilise xscen-0.12 sur doris

Anne-SophieFortin committed:
Update workflow_ESPO-G.py

juliettelavoie commented:
@Anne-SophieFortin Je pense que tu n'as pas encore push ta config ?

Anne-SophieFortin committed:
Update config_ESPO-G_RDRS.yml

juliettelavoie commented on configuration/config_ESPO-G_RDRS.yml:
```diff
@@ -1,10 +1,10 @@
-# Configuration file for ESPO-G6-R2 v1.0.0
+# Configuration file for ESPO-G6E-R3 v1.0.0
```
On va l'appeler C3 et on utilise CaSR même en français parce que personne sait c'est quoi RCaS.
à changer à quelques endroits dans la config.

Anne-SophieFortin committed:
Update config_ESPO-G_RDRS.yml

juliettelavoie commented on configuration/config_ESPO-G_RDRS.yml:
```diff
@@ -110,7 +110,7 @@ extraction:
             allow_conversion: True
             allow_resampling: False
-            restrict_members:
+            restrict_members: # Does restrict_members does the same thing as member in other_search_criteria ?
```
member in `other_search_criteria` you have to name the member you want ( eg. r1i1p1f1). This means take the first one available.

juliettelavoie commented on configuration/config_ESPO-G_RDRS.yml:
```diff
@@ -91,7 +91,7 @@ extraction:
             other_search_criteria:
                 source: &ref_source
-                    RDRS
+                    CaSR
```
en ce moment CaSR 2.1 s'appelle encore RDRS dans les catalogues, mais peut-être que ça va changer.
peut-être safe d'ajouter la version dans les search criterias.

juliettelavoie commented on configuration/config_ESPO-G_RDRS.yml:
```diff
@@ -458,26 +455,26 @@ off-diag:
-      method: bbox
-      lat_bnds:  [ 41.4, 49.4]  #[ 41, 49]
-      lon_bnds: [ -92.2, -59.8 ] #[ -92, -68 ]
+    # Cree-rdrs:
```
tu pourrais aussi juste mettre en commentaire official-diag dans la liste de tâches au début du fichier si tu ne veux pas faire cette tâche

juliettelavoie reviewed (commented):
Looks good to me!

juliettelavoie commented on workflow_ESPO-G.py:
```diff
@@ -10,7 +10,9 @@
 from itertools import product
-from xclim.core.calendar import convert_calendar, get_calendar, date_range_like,doy_to_days_since
+from xclim.core.calendar import get_calendar, doy_to_days_since
+from xarray.core.dataset import convert_calendar
```
est-ce que ça marche vrm ça ? Je pensais que la formulation de la version xarray était un peu différente.
avant: `xclim.convert_calendar(ds,'noleap')`

après (avec xarray): `ds.convert_calendar ('noleap')` (appelle sur l'objet au lieu de l'objet en arg)

Il y aura peut-être quelques lignes de plus à changer dans le workflow pour s'aligner là-dessus.

Vérifie que le calendrier est bien changé.

aulemahal replied:
bonjour, je m'intruse.

Ça marche, mais en effet c'est pas orthodoxe. Règle du pouce pour xarray : si c'est pas direct dans le top du module (ie pas dans le sous-module `xarray.core`) c'est privé et on ne devrait pas l'utiliser. 

J'encourage aussi   `ds.convert_calendar('noleap')`!

Anne-SophieFortin replied:
Quelle serait la bonne façon de procéder? convert_calendar n'est plus dans xclim à cause qu'il se trouve dans xarray, mais il se trouve dans xarray.core...

aulemahal replied:
Utiliser le méthode de classe plutôt que la fonction serait la manière de faire! Donc:
```python3
ds = ds.convert_calendar('noleap')
```

juliettelavoie replied:
Tu enlèves tous les import `convert_calendar` (xclim et xarray). Dans le workflow, quand tu vois un appel à la version xclim, tu changes pour la version xarray.
ex à la ligne 117:


`ds_refnl = convert_calendar(ds_ref, "noleap")` -> `ds_refnl=ds_refnl.convert_calendar('noleap')`

juliettelavoie replied:
oups j'avais pas vu le réponse de Pascal, sorry!

Anne-SophieFortin replied:
Merci à vous deux pour les explications, j'ai fait les modifs.

Anne-SophieFortin committed:
Update config_ESPO-G_RDRS.yml

Anne-SophieFortin committed:
Update workflow_ESPO-G.py

juliettelavoie commented on configuration/config_ESPO-G_RDRS.yml:
```diff
@@ -1,9 +1,9 @@
+    name: ESPO-G6E-C3
     version: 1.0.0
-    description: Ensemble de simulations post-traités d'Ouranos - Global - CMIP6 Étendu - RCaSv3
+    description: Ensemble de simulations post-traités d'Ouranos - Global - CMIP6 Étendu - CaRSv3
```
```suggestion
    description: Ensemble de simulations post-traités d'Ouranos - Global - CMIP6 Étendu - CaRS v3.1
```

Anne-SophieFortin committed:
Update workflow_ESPO-G.py

Utilisation de la méthode de classe plutôt que la fonction convert_calendar

Anne-SophieFortin committed:
Update configuration/config_ESPO-G_RDRS.yml

Co-authored-by: juliettelavoie <juliette.lavoie@hotmail.ca>

Anne-SophieFortin committed:
CaSR possède une grille différente de RDRS. Le domaine de CaSR est rlon: -35.4 à 28.1 & rlat: -44.1 à 25.8.
Cependant, le workflow n'accepte pas que rlon soit négatif. Pour mes tests j'utilise simplement rlon: 1 à 5. Ce sera un problème pour plus tard.

train n'a plus adapt_freq comme argument. Je copie ici la branche ESPO-R et j'ajoute adapt_freq_thresh sous xclim_train_args.

Zeitsperre committed:
update .gitignore

Zeitsperre committed:
update configurations for 2300, added print calls

Anne-SophieFortin committed:
try extract - except extract with cftime=True

Anne-SophieFortin committed:
xr_open_kwargs already defined in config.yml so we need to copy it before adding cftime in the workflow

Anne-SophieFortin committed:
updated indicators for our analysis

Anne-SophieFortin committed:
Added comments and my_indicators

Anne-SophieFortin committed:
will run indicators only for selected models and updated climatology function
